stubs/CoreGenericFunctions.phpstub

```diff
@@ -725,6 +725,26 @@ function join($separator, array $array = []): string
 /**
  * @psalm-pure
  *
+ * @param non-empty-string $separator
+ *
+ * @return (
+ *     $limit is int<min, -2>
+ *         ? array<empty, empty>
+ *         : (
+ *             $limit is int<-1, 1>
+ *                 ? array{string}
```

Comment:
Never mind: this is wrong, since `-1` has the special meaning of "no limit". Will push and amend.